ResourceStatusSystem/Client/ResourceStatus.py

```diff
@@ -114,8 +154,149 @@ def __updateSECache( self ):
+    """
+
+    meta = { 'columns' : [ 'Name', 'StatusType', 'Status' ] }
+
+    for ti in range( 5 ):
+      rawCache = self.rssClient.selectStatusElement( 'Resource', 'Status',
+                                                     elementType = 'ComputingElement',
+                                                     meta = meta )
+      if rawCache['OK']:
+        break
+      self.log.warn( "Can't get CE status", rawCache['Message'] + "; trial %d" % ti )
+      sleep( math.pow( ti, 2 ) )
+      self.rssClient = ResourceStatusClient()
+
+    if not rawCache[ 'OK' ]:
+      return rawCache
+    return S_OK( getCacheDictFromRawData( rawCache[ 'Value' ] ) )
+
+
 ################################################################################
 
+  def __getRSSComputingElementStatus( self, elementName, statusType ):
+    """
+    Gets from the cache or the RSS the ComputingElements status. The cache is a
+    copy of the DB table. If it is not on the cache, most likely is not going
+    to be on the DB.
+
+    There is one exception: item just added to the CS, e.g. new ComputingElement.
+    The period between it is added to the DB and the changes are propagated
+    to the cache will be inconsistent, but not dangerous. Just wait <cacheLifeTime>
+    minutes.
+    """
+
+    cacheMatch = self.ceCache.match( elementName, statusType )
+
+    self.log.debug( '__getRSSComputingElementStatus' )
+    self.log.debug( cacheMatch )
+
+    return cacheMatch
+
+  def __getCSComputingElementStatus( self, elementName, statusType, default ):
+    """
+    Gets from the CS the ComputingElements status
+    """
+
+    cs_path = "/Resources/ComputingElements"
+
+    if not isinstance( elementName, list ):
+      elementName = [ elementName ]
+
+    statuses = self.rssConfig.getConfigStatusType( 'ComputingElement' )
+
+    result = {}
+    for element in elementName:
+
+      if statusType is not None:
+        # Added Active by default
+        res = gConfig.getValue( "%s/%s/%s" % ( cs_path, element, statusType ), 'Active' )
+        result[element] = {statusType: res}
+
+      else:
+        res = gConfig.getOptionsDict( "%s/%s" % ( cs_path, element ) )
+        if res[ 'OK' ] and res[ 'Value' ]:
+          elementStatuses = {}
+          for elementStatusType, value in res[ 'Value' ].items():
+            if elementStatusType in statuses:
+              elementStatuses[ elementStatusType ] = value
+
+          # If there is no status defined in the CS, we add by default Read and
+          # Write as Active.
+          if elementStatuses == {}:
+            elementStatuses = { 'ReadAccess' : 'Active', 'WriteAccess' : 'Active' }
+
+          result[ element ] = elementStatuses
+
+    if result:
+      return S_OK( result )
+
+    if default is not None:
+
+      # sec check
+      if statusType is None:
+        statusType = 'none'
+
+      defList = [ [ el, statusType, default ] for el in elementName ]
+      return S_OK( getDictFromList( defList ) )
```

Comment:
the `__getCSComputingElementStatus` gets the status of a computing element from the CS
the `__getRSSComputingElementStatus` gets the status of a computing element from the RSS